src/components2/MapLegend/MapLegend.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import { breakpoints } from '~/styles/utils';
+
+import LegendExpanded from './LegendExpanded';
+
+interface Props {
+  isPopupVisible: boolean;
+  isDetailOpen: boolean;
+}
+
+const Wrapper = styled.div<Props>`
+  display: ${({ isPopupVisible, isDetailOpen }) =>
+    !isPopupVisible && !isDetailOpen ? 'block' : 'none'};
+
+  @media (min-width: ${breakpoints.m}px) {
+    display: ${({ isDetailOpen }) => (!isDetailOpen ? 'block' : 'none')};
+  }
+`;
+
+const MapLegend = (props: Props) => {
+  // const [isExpanded, setExpanded] = useState(false);
+
+  // on toggle
+  return (
+    <Wrapper {...props}>
```

Comment:
We would always write out props in the method call and in the JSX part. There are a few places where it has more clarity to not do so, but in general, its better to make them explicit.